### PR TITLE
Workflow / Bnb: Add a mechanism to inform us if the import fails

### PR DESCRIPTION
**Pull request by younesbelkada**
As per title and to avoid our CI silently passing

cc @Titus-von-Koeller @BenjaminBossan

### CODE REVIEW

**Review by BenjaminBossan — COMMENTED** [review]
Just for my understanding, this is basically an extra step that checks if bnb import works and reports the results to slack (I wonder if checking `python -m bitsandbytes` would also make sense).

This does not address the issue that the [failing nightly bnb runs](https://github.com/huggingface/peft/actions/runs/9377120709) are not correctly reported on slack, right? Could we also use the `post-slack` workflow for those to report them correctly?

**Comment by younesbelkada** [comment]
Makes sense, added them as suggested ! LMK wdyt !

**Review by BenjaminBossan — APPROVED** [review]
Thanks for working on this. LGTM.

**Comment by Titus-von-Koeller** [comment]
Thanks @younesbelkada for working on this and @BenjaminBossan for raising this and following up!

Everything LGTM as well, but agreed that we should definitely add `python -m bitsandbytes` as a check. I did a quick text search, but didn't find it yet, sorry if I missed it somehow.

File: .github/workflows/nightly-bnb.yml [hunk 1]
```diff
@@ -46,29 +46,85 @@ jobs:
             git checkout tags/v$transformers_version
             cd .. 
           fi
+
+      - name: Test bnb import
+        id: import
+        if: always()
+        run: |
+          python3 -c "import bitsandbytes as bnb"
```
Comment (younesbelkada) [inline .github/workflows/nightly-bnb.yml:54]:
```suggestion
          python -m bitsandbytes
```

**Comment by younesbelkada** [comment]
The issue with `python -m bitsandbytes` is that this will not explictly throw an error if the setup is broken, it will simply display a message saying that bnb is broken, on the other hand importing bnb on the command line will fail and we would be able to catch it with `jobs.import.status`

**Comment by younesbelkada** [comment]
LMK what do you think and I will merge if you think what I said makes sense

**Comment by BenjaminBossan** [comment]
> The issue with `python -m bitsandbytes` is that this will not explictly throw an error if the setup is broken, it will simply display a message saying that bnb is broken, on the other hand importing bnb on the command line will fail and we would be able to catch it with `jobs.import.status`

Can we not have both, first `python -m bitsandbytes` and then importing? That way, we have a bit more information in the logs if something breaks.

File: .github/workflows/nightly-bnb.yml (hunk 1 above)
Comment (younesbelkada) [inline .github/workflows/nightly-bnb.yml:54]:
```suggestion
          python3 -c "import bitsandbytes as bnb"
          python3 -m bitsandbytes
```

**Comment by younesbelkada** [comment]
Done !

**Comment by BenjaminBossan** [comment]
Thanks. Just a small note: Would it make sense to call `python3 -m bitsandbytes` first, in case the import fails the next line is not executed, or is it?

File: .github/workflows/nightly-bnb.yml [hunk 2]
```diff
@@ -46,29 +46,86 @@ jobs:
...
+          python3 -m bitsandbytes
```
Comment (younesbelkada) [inline .github/workflows/nightly-bnb.yml:55]:
```suggestion
          python3 -m bitsandbytes
          python3 -c "import bitsandbytes as bnb"
```

**Comment by younesbelkada** [comment]
makes sense!

**Comment by BenjaminBossan** [comment]
From my point of view, this can be merged, not sure if we want to wait for another look by Titus.

**Comment by younesbelkada** [comment]
just forgot to merge! :D

**Pull request merged**